### PR TITLE
Test on Django 4.2

### PR DESCRIPTION
**Pull request by adamchainz**
Add tests with the new Django LTS release, and tidy up test configuration.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -3,14 +3,13 @@
 envlist = {py36,py39}-{dj32}
           {py38,py310}-{dj40}
           {py38,py310}-{dj41}
-
-toxworkdir = {homedir}/.tox-django-typed-models
```
Comment (adamchainz) [inline tox.ini:7]: The `{homedir}` substitution was removed in tox 4: https://tox.wiki/en/latest/upgrading.html#substitutions-removed

I think we should stick with the default `.tox` directory, I don't see why it was changed. I also added the `.gitignore` rule for `.tox`.

File: tox.ini [hunk 2]
```diff
@@ -3,14 +3,13 @@
...
+          {py38,py310}-{dj42}
 
 [testenv]
 changedir = {toxinidir}
 commands =
     pip --disable-pip-version-check install -e .
-    coverage run {envbindir}/py.test --ds=test_settings typedmodels/tests.py {posargs}
+    coverage run {envbindir}/pytest --ds=test_settings typedmodels/tests.py {posargs}
```
Comment (adamchainz) [inline tox.ini:12]: The pytest command changed to `pytest` in 3.0.0 (2016): https://docs.pytest.org/en/7.1.x/changelog.html#release-3-0-0

(`py.test` still works but best to use the new one)

**Comment by adamchainz** [comment]
I’d like to make a PR for Django 5.0, but I’ll wait for this one to be merged before I put the effort in.

There’s at least one breakage on Django 5.0: `TypedModelMetaclass._patch_fields_cache` needs to update to use the new `get_fields()` signature per https://github.com/django/django/commit/1754c2c8026ab592183821c3933972a8aa7acc66.

**Review by craigds — APPROVED** [review]
looks good, thanks. sorry I ignored this since april

**Pull request merged**